src/Sylius/Component/Registry/DependencyInjection/Compiler/AliasedServicePass.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+
+namespace Sylius\Component\Registry\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Sylius\Component\Registry\ServiceRegistryInterface;
+
+/**
+ * This compiler pass will add services to a registered ServiceRegistry
+ * definition.
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class AliasedServicePass implements CompilerPassInterface
+{
+    private $serviceRegistryId;
+    private $serviceTag;
+    private $options;
+
+    /**
+     * @param string $serviceRegistryId
+     * @param string $serviceTag
+     * @param array $options
+     */
+    public function __construct(
+        $serviceRegistryId,
+        $serviceTag,
+        array $options = []
+    ) {
+        $this->serviceRegistryId = $serviceRegistryId;
+        $this->serviceTag = $serviceTag;
+        $this->options = $options;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition($this->serviceRegistryId)) {
+            return;
+        }
+
+        $options = $this->resolveOptions();
+        $aliasAttribute = $options['alias_attribute'];
+        $context = $options['context'];
+
+        $registry = $container->findDefinition($this->serviceRegistryId);
+        $reflection = new \ReflectionClass($registry->getClass());
+
+        if (!$reflection->implementsInterface(ServiceRegistryInterface::class)) {
+            throw new InvalidArgumentException(sprintf(
+                'The service registry "%s" must implement the "%s" interface.',
+                $this->serviceRegistryId, ServiceRegistryInterface::class
+            ));
+        }
+
+        foreach ($container->findTaggedServiceIds($this->serviceTag) as $id => $attributes) {
+            if (!isset($attributes[0][$aliasAttribute]))  {
```

Comment:
```xml
<service id="foo.bar" class="Foo/Bar">
  <tag name="bar.baz" alias="newAlias" />
  <tag name="bar.baz" alias="legacyAlias" />
</service>
```

This way `$container->findTaggedServiceIds('bar.baz')` will return:

```php
[
  'foo.bar' => [
    ['alias' => 'newAlias'],
    ['alias' => 'legacyAlias'],
  ],
]
```

Right now only the first one will be handled correctly.